### ISSUE TITLE
"AttributeError: '_thread._local' object has no attribute 'type'" when using decord inside ThreadPool

### ISSUE THREAD

**Issue by eawer** [dmlc/decord#85]
I'm trying to use a decord inside a thread pool:
```python
from decord import cpu, gpu
from decord import VideoReader, VideoLoader
from multiprocessing.dummy import Pool as ThreadPool

pool = ThreadPool(1)
pool.map(process_paths, batches)
```
In process_paths I have:
```python
...
vr = VideoReader(video_path, ctx=cpu(0))
print(vr)
print(len(vr))
for i in range(len(vr)):
    frame = vr[i]
...
```
This line `frame = vr[i]` throws:
```
AttributeErrorTraceback (most recent call last)
<timed exec> in <module>

/opt/conda/lib/python3.6/multiprocessing/pool.py in map(self, func, iterable, chunksize)
    264         in a list that is returned.
    265         '''
--> 266         return self._map_async(func, iterable, mapstar, chunksize).get()
    267 
    268     def starmap(self, func, iterable, chunksize=None):

/opt/conda/lib/python3.6/multiprocessing/pool.py in get(self, timeout)
    642             return self._value
    643         else:
--> 644             raise self._value
    645 
    646     def _set(self, i, obj):

/opt/conda/lib/python3.6/multiprocessing/pool.py in worker(inqueue, outqueue, initializer, initargs, maxtasks, wrap_exception)
    117         job, i, func, args, kwds = task
    118         try:
--> 119             result = (True, func(*args, **kwds))
    120         except Exception as e:
    121             if wrap_exception and func is not _helper_reraises_exception:

/opt/conda/lib/python3.6/multiprocessing/pool.py in mapstar(args)
     42 
     43 def mapstar(args):
---> 44     return list(map(*args))
     45 
     46 def starmapstar(args):

<ipython-input-63-c19fac1d9d04> in process_paths(batch)
     40 def process_paths(batch):
     41     for path, model_id in batch:
---> 42         process_path(str(path), model_id)

<ipython-input-63-c19fac1d9d04> in process_path(path, model_id)
     21         bboxes = []
     22         for i in range(len(vr)):
---> 23             frame = vr[i]
     24             print(frame.shape)
     25             print(4.4)

~/.local/lib/python3.6/site-packages/decord-0.4.0-py3.6.egg/decord/video_reader.py in __getitem__(self, idx)
     90             raise IndexError("Index: {} out of bound: {}".format(idx, self._num_frame))
     91         self.seek_accurate(idx)
---> 92         return self.next()
     93 
     94     def next(self):

~/.local/lib/python3.6/site-packages/decord-0.4.0-py3.6.egg/decord/video_reader.py in next(self)
    105         if not arr.shape:
    106             raise StopIteration()
--> 107         return bridge_out(arr)
    108 
    109     def _validate_indices(self, indices):

~/.local/lib/python3.6/site-packages/decord-0.4.0-py3.6.egg/decord/bridge/__init__.py in bridge_out(native_arr)
     31 
     32 def bridge_out(native_arr):
---> 33     return _BRIDGE_TYPES[_CURRENT_BRIDGE.type][0](native_arr)
     34 
     35 def bridge_in(arr):

AttributeError: '_thread._local' object has no attribute 'type'
```

**Comment by eawer** [comment]
hey, @zhreshold , sure
```python
from decord import cpu, gpu
from decord import VideoReader, VideoLoader
from multiprocessing.dummy import Pool as ThreadPool

video_paths = [
  #list of paths to video
]

def process_path(path):
    vr = VideoReader(path, ctx=cpu(0))

    for i in range(len(vr)):
        frame = vr[i]

pool = ThreadPool(1)
pool.map(process_path, video_paths)
```

**Comment by zhreshold** [comment]
This should be fixed now after #86 is merged
